### PR TITLE
[polaris.shopify.com] Clean up example titles

### PR DESCRIPTION
**Pull request by alex-page**
- [x] Removed component names from example titles as they are on the page title
- [x] Removed `simple` or `basic` from example titles
- [x] Updated README, content and examples dir
- [x] Fixed markdown issues when investigating migration
- [x] Tophatted all examples and logged issues in https://github.com/Shopify/polaris/issues/6706

### CODE REVIEW

File: .changeset/odd-wolves-attack.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'polaris.shopify.com': minor
+---
+
+Example descriptions support for markdown content
```
Comment (martenbjork) [inline .changeset/odd-wolves-attack.md]: You have 2 changesets in this PR

File: polaris-react/src/components/Card/README.md [hunk 2]
```diff
@@ -450,7 +450,7 @@ Use to render custom content such as icons, links, or buttons in a card section
 </Card>
 ```
 
-### Card with all of its elements
+### With all of its elements
```
Comment (martenbjork) [inline polaris-react/src/components/Card/README.md]: Maybe even:

> With all elements

**Review by martenbjork — APPROVED** [review]

File: polaris-react/src/components/VideoThumbnail/README.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ Video thumbnails should:
 
 ## Examples
 
-### Basic video thumbnail
+### Default
```
Comment (martenbjork) [inline polaris-react/src/components/VideoThumbnail/README.md:36]: Help a tired person out: Why do we not delete these readmes? Aren't they 1:1 with what's in the website directory? Is that part of your migration strategy?

**Review by martenbjork — APPROVED** [review]

File: polaris-react/src/components/VideoThumbnail/README.md (hunk 3 above)
Comment (alex-page) [reply polaris-react/src/components/VideoThumbnail/README.md:36]: I did this so I could make sure we didn't lose any content as it hadn't been migrated across in a long time. We definitely should delete them but first I need to make sure the content is not lost.

Comment (martenbjork) [reply polaris-react/src/components/VideoThumbnail/README.md:36]: Gotcha!

File: .changeset/odd-wolves-attack.md (hunk 1 above)
Comment (martenbjork) [reply .changeset/odd-wolves-attack.md]: Now 3 🤔

**Pull request merged**